ledger/complete/ledger.go

```diff
@@ -368,3 +376,26 @@ func (l *Ledger) DumpTrieAsJSON(state ledger.State, outputFilePath string) error
 
 	return trie.DumpAsJSON(writer)
 }
+
+// this operation should only be used for exporting
+func (l *Ledger) keepOnlyOneTrie(state ledger.State) error {
+	// don't write things to WALs
+	l.wal.PauseRecord()
+	defer l.wal.UnpauseRecord()
+
+	allTries, err := l.forest.GetTries()
+	if err != nil {
+		return err
+	}
+
+	targetRootHash := ledger.RootHash(state)
+	for _, trie := range allTries {
+		trieRootHash := trie.RootHash()
+		if !bytes.Equal(trieRootHash, targetRootHash) {
+			l.forest.RemoveTrie(trieRootHash)
+		}
+	}
+	// call gc to free up mem
+	runtime.GC()
```

Comment:
is this a good measure? @zhangchiqing

Reply:
Probably, but I would say let's hold off until we do some benchmark.

Reply:
This line alone forces GC to be triggered, which comes with a cost that takes some CPU time, and might actually decrease the performance.

Note that, releasing releasable memory will not improve performance, but just allow us to make room for allocating more memories without hitting out of memory. 

To improve performance, we would like to spend less CPU on GC by triggering it less often and make sure it's only triggered when necessary and before hitting out of memory.

We probably want to manually trigger GC at this point, since it's likely the moment a lot of memory garbage are just generated and can be freed. However, if we decide to manually trigger GC, then we would need to first turn off the automatic GC by setting `GOGC=OFF`, which will give up some other default trigger conditions. 

Given we are still understanding our GC schedules on actual environment, I would suggest to hold off for now until we did some benchmark and see what are the best GC-trigger conditions.

Reply:
I think thats the goal, note that this is only used in the spork prep step.